sympy/functions/elementary/piecewise.py

```diff
@@ -371,10 +370,7 @@ def _eval_simplify(self, ratio, measure, rational, inverse):
                     for e in eqs:
                         # these blessed lhs objects behave like Symbols
                         # and the rhs are simple replacements for the "symbols"
-                        if isinstance(e.lhs, (Symbol, UndefinedFunction)) and \
-                            isinstance(e.rhs,
-                                (Rational, NumberSymbol,
-                                Symbol, UndefinedFunction)):
+                        if _blessed(e):
```

Comment:
I've been working on https://github.com/sympy/sympy/issues/19484 and wanted to ask what significance does this condition have here. As it checks on the basis of `_diff_wrt` there are some cases where Piecewise is not simplified as it set to `False` for instances of `Expr`